doc/dev/transforms.rst

```diff
@@ -100,21 +98,154 @@ classical instructions as well as side-effecting operations (e.g., print) can al
+
+.. code-block:: cpp
+
+    #define DEBUG_TYPE "myhelloworld"
+
+    #include "Catalyst/IR/CatalystDialect.h"
+    #include "mlir/Pass/Pass.h"
+    #include "llvm/Support/Debug.h"
+
+    using namespace llvm;
+    using namespace mlir;
+    using namespace catalyst;
+
+    namespace catalyst {
+    #define GEN_PASS_DEF_MYHELLOWORLDPASS
+    #define GEN_PASS_DECL_MYHELLOWORLDPASS
+    #include "Catalyst/Transforms/Passes.h.inc"
+
+    struct MyHelloWorldPass : public impl::MyHelloWorldPassBase<MyHelloWorldPass> {
+        using impl::MyHelloWorldPassBase<MyHelloWorldPass>::MyHelloWorldPassBase;
+
+        void runOnOperation() override { llvm::errs() << "Hello world!\n"; }
+    };
+
+    std::unique_ptr<Pass> createMyHelloWorldPass() { return std::make_unique<MyHelloWorldPass>(); }
+
+    } // namespace catalyst
+
+We make the pass object ``MyHelloWorldPass``, which inherits from the base class ``MyHelloWorldPassBase`` that tablegen will build in the namespace ``impl``. 
+The function that determines what your pass actually does is the ``void runOnOperation()``. Here all the pass does is print out ``"Hello world!\n"``. 
+
+(A sidenote on printing messages in MLIR: there are two major printing options in LLVM. The `more standard one <https://llvm.org/docs/ProgrammersManual.html#the-llvm-debug-macro-and-debug-option>`_ is ``dbgs()``, which only prints when a debug flag is set. 
+The other option is the ``errs()`` used here, which will print no matter what.)
+
+This new C++ file needs to be added to the ``mlir/lib/Catalyst/Transforms/CMakeLists.txt`` file (or the CMakeLists.txt of whichever directory that has your new pass file): 
+
+.. code-block::
+
+    file(GLOB SRC
+        ...
+        MyHelloWorldPass.cpp
+    )
+
+After writing the pass, we need to register it in a few places. In ``mlir/include/Catalyst/Transforms/Passes.h``, add the method 
+
+.. code-block:: cpp
+
+    namespace catalyst {
+        ...
+        std::unique_ptr<mlir::Pass> createMyHelloWorldPass();
+        ...
+    }
+
+And in ``mlir/lib/Catalyst/Transforms/RegisterAllPasses.cpp``, register the pass via 
+
+.. code-block:: cpp
+
+    void catalyst::registerAllCatalystPasses()
+    {
+        ...
+        mlir::registerPass(catalyst::createMyHelloWorldPass);
+        ...
+    }
+
+Note that this addition in ``RegisterAllPasses.cpp`` needs to happen in the ``lib/Catalyst/Transforms`` directory, regardless of which dialect your pass belongs to.
+
+Now that we have written our shiny new pass, we can build it by going back to the top-level ``catalyst`` directory and 
+
+.. code-block::
+
+    make dialects
+
+The tool to run passes is built as ``mlir/build/bin/quantum-opt``. 
+Since this is an executable, it needs to be invoked as ``./quantum-opt`` instead of just plain ``quantum-opt`` (if you are in the ``mlir/build/bin`` directory; otherwise supply the full path).
```

Comment:
Maybe we can add that in the .zshrc we can add the line:

```
export PATH=.../mlir/build/bin:$PATH
```
where ... is the the Catalyst directory location.

Reply:
done